forwardauth.go

```diff
@@ -199,9 +199,16 @@ func (f *ForwardAuth) redirectBase(r *http.Request) string {
 
 // Return url
 func (f *ForwardAuth) returnUrl(r *http.Request) string {
+	replacedPath := r.Header.Get("X-Replaced-Path")
+	if replacedPath != "" {
+		// request-modifier and path-related matchers cannot be used at the same time
+		// so if X-Replaced-Path is set that means X-Forwarded-Prefix et al. are not
+		return fmt.Sprintf("%s%s", f.redirectBase(r), replacedPath)
+	}
+	prefix := r.Header.Get("X-Forwarded-Prefix")
 	path := r.Header.Get("X-Forwarded-Uri")
 
-	return fmt.Sprintf("%s%s", f.redirectBase(r), path)
+	return fmt.Sprintf("%s%s%s", f.redirectBase(r), prefix, path)
```

Comment:
If Traefik is configured through Kubernetes ingress handlers, and the router is `/a/b/` and the path requested is `/a/b/`, then `traefik-forward-auth` will receive
```
X-Forwarded-Prefix: /a/b/
X-Forwarded-Uri: /
```

I found that setting `path` using this function works:
```
func GetUriPath(r *http.Request) string {
	prefix := r.Header.Get("X-Forwarded-Prefix")
	uri := r.Header.Get("X-Forwarded-Uri")
	p := path.Join(prefix, uri)
	if strings.HasSuffix(uri, "/") {
		p += "/"
	}
	return p
}
```

Reply:
we whould not add another / at the end of the query if there is already one.

Reply:
Under the circumstances that I have described, the `fmt.Sprintf` adds two slashes to the end.

With the code I've suggested, the `path.Join` always strips trailing slashes, so we add it back if it is in the original URL. It only leaves one trailing slash at the most.